### ISSUE TITLE
Call deferred.cancel() only once when cleaning the reactor

### ISSUE THREAD

**Issue by chevah-robot** [chevah/compat#612]
T1069 task was created by alibotean on 2013-02-06 12:23:46Z.
Last changed on 2013-02-11 10:16:00Z.

We call the cancel() method of the delayed calls from two distinct places:
 * when stopping the reactor
 * on tearDown() in the testcase

We should call the method only once and from the Twisted reactor thread just as below:

```#!python
def cancel_calls():
    for p in reactor.getDelayedCalls():
        if p.active():
            p.cancel()

reactor.callFromThread(cancel_calls)
```

This will help us get rid of the ```AlreadyCancelled``` exceptions when running the tests.
